### ISSUE TITLE
[BUG] TsGroup with empty value results loss of metadata after saving to .npz

### ISSUE THREAD

**Issue by qian-chu** [pynapple-org/pynapple#252]
When a TsGroup has any of its value being an empty array (e.g. no spike in one trial during PETH calculation), Pynapple assigns an NaN to `rate` metadata. However, after saving the TsGroup to .npz and reloading again using `nap.load_file()`, the entry becomes missing and the metadata dataframe will also lose all columns except `rate`. Some simple codes to replicate the problem:

```
import numpy as np
import pandas as pd
import pynapple as nap

meta_df = pd.DataFrame({'a': [1, 2, 3], 'b': ['xxx', 'yyy', 'zzz']})
my_ts = {
    0: nap.Ts(t=np.sort(np.random.uniform(0, 100, 1000)), time_units="s"),
    1: nap.Ts(t=np.sort(np.random.uniform(0, 100, 2000)), time_units="s"),
    2: nap.Ts(t=np.sort(np.random.uniform(0, 100, 3000)), time_units="s"),
    }
tsgroup1 = nap.TsGroup(my_ts)
tsgroup1.set_info(meta_df)
tsgroup1.save('tsgroup1.npz')
print(tsgroup1)

my_ts[1] = nap.Ts(t=np.array([]), time_units="s")
tsgroup2 = nap.TsGroup(my_ts)
tsgroup2.set_info(meta_df)
tsgroup2.save('tsgroup2.npz')
print(tsgroup2)

tsgroup1_new = nap.load_file('tsgroup1.npz')
print(tsgroup1_new)

tsgroup2_new = nap.load_file('tsgroup2.npz')
print(tsgroup2_new)
```

Output:
```
  Index    rate    a  b
-------  ------  ---  ---
      0   10       1  xxx
      1   20       2  yyy
      2   30.01    3  zzz
  Index    rate    a  b
-------  ------  ---  ---
      0   10       1  xxx
      1  nan       2  yyy
      2   30.01    3  zzz
  Index    rate  b      a
-------  ------  ---  ---
      0   10     xxx    1
      1   20     yyy    2
      2   30.01  zzz    3
  Index    rate
-------  ------
      0   10
      2   30.01
```

**Comment by gviejo** [comment]
Thanks for this issue. It's fixed in #255. I will bump a new version this week. If you need it now, you can install pynapple from the github repository.

**Comment by qian-chu** [comment]
This is fantastic! Thanks for the fast fix.
